src/realm/table.cpp

```diff
@@ -650,6 +650,15 @@ void Table::init(ref_type top_ref, ArrayParent* parent, size_t ndx_in_parent, bo
 
 ColKey Table::do_insert_column(ColKey col_key, DataType type, StringData name, Table* target_table, DataType key_type)
 {
+    // Outgoing links from an asymmetric table are not allowed.
```

Comment:
we will also need exceptions thrown when attempting to set a mixed value which is a TypedLink